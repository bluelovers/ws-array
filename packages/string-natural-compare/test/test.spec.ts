import naturalCompare, { createNew } from '../core';

test(`number`, () =>
{
	let arr = [1, 5, 4];

	let actual = arr.sort(naturalCompare);

	expect(actual).toStrictEqual([1, 4, 5]);

	expect(actual).toMatchSnapshot();

});

test(`desc`, () =>
{
	let arr = [1, 5, '4'];

	let actual = arr.sort(createNew({
		desc: true,
	}));

	expect(actual).toStrictEqual([5, '4', 1]);

	expect(actual).toMatchSnapshot();

});
