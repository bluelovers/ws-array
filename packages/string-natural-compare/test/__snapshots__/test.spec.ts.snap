// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`desc 1`] = `
[
  5,
  "4",
  1,
]
`;

exports[`number 1`] = `
[
  1,
  4,
  5,
]
`;
